### ISSUE TITLE
Bootstrap process fails to download binary from s3 using aws-cli

### ISSUE THREAD

**Issue by schrill** [awslabs/amazon-eks-ami#210]
**What happened**: Bootsrap process fails in file install-worker.sh on line 125

**What you expected to happen**: It should have downloaded binary files

**How to reproduce it (as minimally and precisely as possible)**: just git clone and execute make

**Anything else we need to know?**: Part from output with echo placed to verify the execute part

    amazon-ebs: Downloading binaries from: s3://amazon-eks
    amazon-ebs: AWS cli present - using it to copy binaries from s3.
    amazon-ebs: aws s3 cp --region us-west-2 s3://amazon-eks/1.11.5/2018-12-06/bin/linux/amd64/kubelet .
    amazon-ebs: aws s3 cp --region us-west-2 s3://amazon-eks/1.11.5/2018-12-06/bin/linux/amd64/kubelet.sha256 .
    amazon-ebs: fatal error: An error occurred (400) when calling the HeadObject operation: Bad Request

**Environment**:
- AWS Region: default (taken from git no edits), eu-west-1
- Instance Type(s): default (taken from git no edits)
- AMI Version: region default
- Kernel (e.g. `uname -a`): ami default

**Comment by sie-dlanz** [comment]
Binary downloads are now failing with 403 Forbidden.

**Comment by kinghajj** [comment]
This is almost certainly due to #175. It changed the packer builder definition to include environment variables for AWS_ACCESS_KEY_ID & AWS_SECRET_ACCESS_KEY. The problem is, that if you don't specify the packer variables for these, they'll be blank; and, even so, the mere fact that they exist means that the `aws` CLI will only consider them, and not try other credentials in the chain. One solution might be to detect if those env vars are blank in install-worker.sh, and unset them.

**Comment by teancom** [comment]
I'm not sure if this is related, but even with #212, I'm still failing to download the binaries. When I run `wget` manually, I get the following:

```
 wget 'https://amazon-eks.s3.amazonaws.com/amazon-eks/1.11.5/2018-12-06/bin/linux/amd64/aws-iam-authenticator' --2019-03-07 20:48:19--  https://amazon-eks.s3.amazonaws.com/amazon-eks/1.11.5/2018-12-06/bin/linux/amd64/aws-iam-authenticator
Resolving amazon-eks.s3.amazonaws.com (amazon-eks.s3.amazonaws.com)... 52.218.248.218
Connecting to amazon-eks.s3.amazonaws.com (amazon-eks.s3.amazonaws.com)|52.218.248.218|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2019-03-07 20:48:19 ERROR 404: Not Found.
```

Note: I also had to prepend `amazon-eks` to `$S3_URL_BASE` to fix a problem where it was returning a `301` error code with no destination:

```
wget 'https://s3.amazonaws.com/amazon-eks/1.11.5/2018-12-06/bin/linux/amd64/aws-iam-authenticator' 
--2019-03-07 20:51:07--  https://s3.amazonaws.com/amazon-eks/1.11.5/2018-12-06/bin/linux/amd64/aws-iam-authenticator
Resolving s3.amazonaws.com (s3.amazonaws.com)... 54.231.33.162
Connecting to s3.amazonaws.com (s3.amazonaws.com)|54.231.33.162|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: unspecified
ERROR: Redirection (301) without location.
```

**Comment by teancom** [comment]
If it matters, I'm using `us-east-1`, which has some special logic in `install-worker.sh`

**Comment by micahhausler** [comment]
This should be fixed by #212. Can you try again with that change?

**Comment by teancom** [comment]
Sorry, was unrelated. I made a change to the json config file that didn't work out the way I'd hoped :-) We're all good, thanks.
